### PR TITLE
Wolmo Toast

### PR DESCRIPTION
**Pull request by sbalay**
## Summary
Wolmo toast component.

#### Android
wolmo toast android

#### Ios
wolmo toast

### CODE REVIEW

File: src/WMOToast/WMOToast.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    return (
+      <Text>
+        {message}
+      </Text>
+    );
+  }
+};
+
+WMOToast.propTypes = {
+  containerStyle: React.PropTypes.any, // eslint-disable-line react/forbid-prop-types
```
Comment (mvbattan) [inline src/WMOToast/WMOToast.js]: Styles should be passed via `View.propTypes.style` or `Text.propTypes.style`

File: src/WMOToast/WMOToastRedux.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+export const actions = stringArrayToObject(['DISPLAY_ERROR', 'DISPLAY_WARNING', 'DISPLAY_INFO', 'HIDE']);
+
+export const actionCreators = {
+  hide() {
+    return {
+      type: actions.HIDE
+    };
+  },
+  displayError(message, duration = 4000) {
```
Comment (mvbattan) [inline src/WMOToast/WMOToastRedux.js]: I'd use a common aux function for `displayError`, `displayWarning` and `displayInfo`.

File: src/WMOToast/WMOToastRedux.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    };
+  }
+};
+
+/* ------------- Toast reducer ------------- */
+
+const defaultState = {
+  message: null,
+  error: false,
+  warning: false,
```
Comment (mvbattan) [inline src/WMOToast/WMOToastRedux.js]: Comment: IMHO `error` and `warning` might be hard to maintain later, especially if someone wishes to create a custom toast (e.g. changing its background color and font).
Maybe an replacing enum would help here.

File: src/WMOToast/WMOToast.js [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+  hide() {
+    Animated.timing(this.state.shadowOpacity, { toValue: 0 }).start();
+    Animated.timing(this.state.fadeAnimation, { toValue: 0 }).start(() => {
+      this.setState({ present: false, message: null, error: false, warning: false, dismissTimeout: null });
+    });
+  }
+
+  render() {
+    if (this.state.present) {
```
Comment (mvbattan) [inline src/WMOToast/WMOToast.js]: `if (!this.state.present) return null` to avoid code indentation

File: src/WMOToast/WMOToastContainer.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import { connect } from 'react-redux';
+
+import WMOToast from './WMOToast';
+
+const mapStateToProps = store => ({
+  message: store.toast.message,
+  error: store.toast.error,
+  warning: store.toast.warning,
+  duration: store.toast.duration
```
Comment (mvbattan) [inline src/WMOToast/WMOToastContainer.js]: `const mapStateToProps = store => { ...store.toast }` ?

File: src/WMOToast/WMOToastRedux.js [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+          warning: false
+        },
+        { deep: true }
+      );
+    }
+    case actions.HIDE: {
+      return state.merge(
+        {
+          message: null,
+          duration: null,
```
Comment (mvbattan) [inline src/WMOToast/WMOToastRedux.js]: If `HIDE` contains `message: null` and `duration: null` in its payload, we can compress this reducer into:
```
case actions.HIDE:
case actions.DISPLAY_INFO:
case actions.DISPLAY_WARNING:
case actions.DISPLAY_ERROR:
    return state.merge({
        message: action.payload.message,
        duration: action.payload.duration,
        error: action.type === actions.DISPLAY_ERROR,
        warning: action.type === actions.DISPLAY_WARNING
     },
        { deep: true }
     );
```

File: src/WMOToast/WMOToastRedux.js [hunk 7]
```diff
@@ -0,0 +1,103 @@
+import Immutable from 'seamless-immutable';
+
+import { stringArrayToObject } from '../redux/reduxUtils';
+
+/* ------------- Toast actions ------------- */
+
+export const actions = stringArrayToObject(['DISPLAY_ERROR', 'DISPLAY_WARNING', 'DISPLAY_INFO', 'HIDE']);
```
Comment (mvbattan) [inline src/WMOToast/WMOToastRedux.js]: Use `@@WMOToast/ActionName` for every action name.

File: src/WMOToast/WMOToastRedux.js (hunk 3 above)
Comment (sbalay) [inline src/WMOToast/WMOToastRedux.js]: `warning` and `error` are pretty standard words for logging and showing alerts. I'd keep them.
To customize the toast you can use the style props that the container have.
What do you think?

**Comment by sbalay** [comment]
@mvbattan ready

File: src/WMOToast/WMOToastRedux.js (hunk 3 above)
Comment (mvbattan) [inline src/WMOToast/WMOToastRedux.js]: Let's keep them then 😁

**Pull request merged**